add instructions for configurable istio access logs (#13239)
kyma-project/kyma: docs/04-operation-guides/operations/obsv-03-configurable-istio-access-logs.md

---
title: Configure Istio access logs
---

[Istio access logs](https://istio.io/latest/docs/tasks/observability/logs/access-log/) can be enabled on a selective basis to provide fine-grained details which can help in indicating the golden signals, and troubleshoot anomalies.

>**Disclaimer:** Enabling access logs may increase logs volume drastically and might fill up your log storage in a very short time. Also, the provided feature makes use of an API in alpha state which may or may not be continued in future releases.

## Prerequisites

The following resources are needed to observe access logs in you CLS setup:

- A CLS instance to ship the logs to.
- Access to the Elasticsearch API enabled on the above CLS instance.

## Configuration

> ### Entire Namespace

Reviewer: ```suggestion
### Configure Istio access logs for the entire Namespace
```

Replace `your-namespace` in the below configuration with your namespace. Use `kubectl apply` for applying the configuration.

```yaml
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: access-config
  namespace: "your-namespace"
spec:
  accessLogging:
    - providers:
      - name: envoy
```

> ### Selective Workload

Reviewer: ```suggestion
###  Configure Istio access logs for a selective workload
```

Label-based selection of workloads can be configured by using [selector](https://istio.io/latest/docs/reference/config/type/workload-selector/#WorkloadSelector).

> Replace `your-namespace` and `your-label` in the below configuration with your namespace and label of workload respectively.

Reviewer: ```suggestion
1. In the following sample configuration, replace {YOUR_NAMESPACE} and {YOUR_LABEL} with your Namespace and the label of the workload, respectively.
```

> Use `kubectl apply` for applying the configuration.

Reviewer: ```suggestion
2. To apply the configuration, run `kubectl apply`.
```

```yaml
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: access-config

>   namespace: "your-namespace"

Reviewer: ```suggestion
  namespace: {YOUR_NAMESPACE}
```

spec:
  selector:
    matchLabels:

>       service.istio.io/canonical-name: "your-label"

Reviewer: ```suggestion
      service.istio.io/canonical-name: {YOUR_LABEL}
```
